tests/test_virtualenv.py

```diff
@@ -279,6 +279,66 @@ def test__resolved_interpreter_windows_pyexe_fails(sysfind, make_one):
     sysfind.assert_any_call("py")
 
 
+@pytest.mark.parametrize(
+    ["input_", "path_result", "expected"],
+    [
+        ("3.7", r"c:\python37-x64\python.exe", r"c:\python37-x64\python.exe"),
+        ("3.7", r"c:\fails\python.exe", None),
+        ("python3.7", r"c:\python37-x64\python.exe", None),
+        ("2.7", r"c:\python37-x64\python.exe", None),
+        ("goofy", r"c:\python37-x64\python.exe", None),
+        ("3.7", None, None),
+        ("python3.7", None, None),
+        ("2.7", None, None),
+        ("goofy", None, None),
+    ],
+)
+@mock.patch("nox.virtualenv._SYSTEM", new="Windows")
+@mock.patch.object(py.path.local, "sysfind")
+def test__resolved_interpreter_windows_path_and_version(
+    sysfind, make_one, input_, path_result, expected
+):
+    # Establish that if we get a standard pythonX.Y path, we look it
+    # up via the path on Windows.
+    venv, _ = make_one(interpreter=input_)
+
+    # Trick the system into thinking that it cannot find python3.6
+    # (it likely will on Unix). Also, we don't give it a dummy
+    # py launcher. But we give it a dummy python interpreter to find
+    # in the system path.
+    if path_result:
```

Comment:
There are 4 if statements and 2 closures in this test case. I'd much rather see it broken up into distinct test cases with a "single path of execution" to make it clear (and hopefully :crossed_fingers: more concise) what is happening in each.

Reply:
Yes, not my most elegant test case indeed. 😬 I tried to split it up earlier and found that I would have to repeat a lot preparation code (creating mocks and making sure the checks above the check-under-test fail). I *could* create yet another fixture to help with this, but I'm not sure whether this makes it even less transparent. I'm unsure, should I choose conciseness or clarity in this case?

Reply:
Try to break this up without any `if` statements? If it gets too unwieldy we can revert?

Reply:
Thanks for all your comments. I pushed my changes just now. Most of the branching actually was part of patching/mocking sysfind, the resulting path object, and python-sysexec the right way for the various parameters. Off-loading the mocking and patching into two pytest fixtures solved the complexity problem (and as a side-effect, created fixtures that are probably useful for other tests in the module as well).